src/service.ts

```diff
@@ -0,0 +1,21 @@
+import {fromFetch} from "rxjs/fetch";
+import {catchError, map, of, switchMap, take, tap} from "rxjs";
+
+const flightData$ = fromFetch("https://opensky-network.org/api/states/all").pipe(
+    take(30),
```

Comment:
This doesn't filter, it actually will take the first 30 emissions of events from the `fromFetch`, then self-destruct.
Rather use `filter()` within the subscribe.